### DISCUSSION TITLE
Groups of cells groups of vertices

### DISCUSSION THREAD

**Discussion by nicolas-tardieu** [pyvista/pyvista#1481] [Q&A] [answered]
Dear pyvista Gurus,

In the context of the Finite Element Method, it is often handy to define groups of vertices or groups of cells in order to apply boundary conditions or assign different materials to different parts of a domain. Once done, it is also very useful to only display given groups or to color them differently one from the other. This way, the user can check its assignations to the domain he is modelling.
Is there this kind of feature in pyvista ? I have the feeling that the Mesh data-structure does not offer these attributes but perhaps do I consider the problem from a wrong point of view.

---

Sorry for coming back to my question. I think it is strongly linked to
pyvista/pyvista-support#267 (comment)
I wonder if cell_arrays and point_arrays are the answer to my answer.
Can someone confirm this ?

Best regards,
Nicolas

**Comment by akaszynski** [comment, marked as answer]
That's exactly what you're looking for.  See:
- [point_arrays](https://docs.pyvista.org/core/dataset.html?highlight=point_arrays#pyvista.DataSet.point_arrays)
- [cell_arrays](https://docs.pyvista.org/core/dataset.html?highlight=cell_arrays#pyvista.DataSet.cell_arrays)
- [Basic API Usage](https://docs.pyvista.org/user-guide/simple.html?highlight=cell_arrays)

I also use `pyvista` heavily in finite element analysis (pre and post-processing).  Point and cell arrays are really convenient ways of organizing the point and cell data.

---

FYI - I've transferred this issue to a discussion.  We're in the process of migrating our Q&A support to discussions so users only have one repo to look for general information or posing questions that don't fit into the standard "issue" format.

**Reply by kskollepara** [reply]
Do the methods `pyvsita.DataSet.point_arrays` and `pyvista.DataSet.cell_arrays` still exist, or are they deprecated? I can't seem to find them in current documentation. Also, the links given in the answer are dead.

**Reply by MatthewFlamm** [reply]
It is now `point_data` and `cell_data`.

**Reply by kskollepara** [reply]
thank you!
